docs/examples/tutorial_signal_processing.py

```diff
@@ -0,0 +1,397 @@
+# -*- coding: utf-8 -*-
+"""
+Computing Wavelet Transform
+============
+This tutorial demonstrates how we can use the signal processing tools within Pynapple to aid with data analysis.
+We will examine the dataset from [Grosmark & Buzsáki (2016)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4919122/).
+
+Specifically, we will examine Local Field Potential data from a period of active traversal of a linear track.
+
+This tutorial was made by Kipp Freud.
+
+"""
+
+
+# %%
+# !!! warning
+#     This tutorial uses matplotlib for displaying the figure
+#
+#     You can install all with `pip install matplotlib requests tqdm seaborn`
+#
+# First, import the necessary libraries:
+
+import math
+import os
+
+import matplotlib.pyplot as plt
+import numpy as np
+import requests
+import seaborn
+import tqdm
+
+seaborn.set_theme()
+
+import pynapple as nap
+
+# %%
+# ***
+# Downloading the data
+# ------------------
+# Let's download the data and save it locally
+
+path = "Achilles_10252013_EEG.nwb"
+if path not in os.listdir("."):
+    r = requests.get(f"https://osf.io/2dfvp/download", stream=True)
+    block_size = 1024 * 1024
+    with open(path, "wb") as f:
+        for data in tqdm.tqdm(
+            r.iter_content(block_size),
+            unit="MB",
+            unit_scale=True,
+            total=math.ceil(int(r.headers.get("content-length", 0)) // block_size),
+        ):
+            f.write(data)
+
+
+# %%
+# ***
+# Loading the data
+# ------------------
+# Let's load and print the full dataset.
+
+data = nap.load_file(path)
+FS = 1250
+print(data)
+
+
+# %%
+# ***
+# Selecting slices
+# -----------------------------------
+# We will consider a single run of the experiment - where the rodent completes a full traversal of the linear track,
+# followed by 4 seconds of post-traversal activity.
+
+# Define the run to use for this Analysis
+run_index = 7
+# Define the IntervalSet for this run and instantiate both LFP and
+# Position TsdFrame objects
+RUN_interval = nap.IntervalSet(
+    data["forward_ep"]["start"][run_index],
+    data["forward_ep"]["end"][run_index] + 4.0,
+)
+RUN_Tsd = data["eeg"].restrict(RUN_interval)
+RUN_pos = data["position"].restrict(RUN_interval)
+print(RUN_Tsd)
+
+# %%
+# ***
+# Plotting the LFP and Behavioural Activity
+# -----------------------------------
+
+fig = plt.figure(constrained_layout=True, figsize=(10, 6))
+axd = fig.subplot_mosaic(
+    [["ephys"], ["pos"]],
+    height_ratios=[1, 0.4],
+)
+
+axd["ephys"].plot(
+    RUN_Tsd[:, 0].restrict(
+        nap.IntervalSet(
+            data["forward_ep"]["start"][run_index], data["forward_ep"]["end"][run_index]
+        )
+    ),
+    label="Traversal LFP Data",
+    color="green",
+)
+axd["ephys"].plot(
```

Comment:
No need for that. It doens't bring much information. Plus it's very artificial when the run stops and starts. Plot the LFP with a single color. The bottom plot help to show when the animal stops running.